### PR TITLE
[IMP][8.0][22070][ISSUE#115]-Modify projects and BR id relationship

### PR DESCRIPTION
**Pull request by serpentcs-dev1**
Issue #115

- Added Smart button Linked Project.

- Added two fields business_requirement_id and business_requirement_deliverable_id in project.project.

- Make invisible linked_project field form BR.

-  Made read-only field for_br while generated projects wizard is open.

@seb-elico @elicoidal @victormartinelicocorp

### CODE REVIEW

**Comment by elicoidal** [comment]
@darshan-serpent thanks for the contribution.
I made a test with one BR, 2 deliverables and created the projects from both BR + deliverables:
- [x] In created project: cannot use the BR smart button to display the linked BR (nothing listed)
- [x] in original BR, Linked Project smart button only shows the project directly linked to the BR, not the ones linked to deliverables (expected shows all projects linked to the BR and its deliverable)

Rest looks good but I still need a test on BR with sub-BR
cc @seb-elico @victormartinelicocorp

**Comment by elicoidal** [comment]
Test with Children project are similar to the one with Deliverables

Problem while copying:
- [x] when copying a BR, the copied BR is still linked to the project of the initial BR (??)

**Comment by serpentcs-dev1** [comment]
@elicoidal Regarding creating projects from Deliverables.

I am not really sure how to create a project from deliverables, can you please describe it in brief?!

**Comment by elicoidal** [comment]
@darshan-serpent What is the situation on this BR?

**Comment by elicoidal** [comment]
@darshan-serpent @sudhir-serpentcs What is the situation here?

**Comment by serpentcs-dev1** [comment]
@elicoidal There's still some confusion here, I'll let you know the details soon of current situation.

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal @victormartinelicocorp 
Could you please review it.

**Comment by elicoidal** [comment]
Small details:

- [ ] in smart button BR in project: the number of BR should not be displayed (a project can only be linked to one BR at a time)
- [x] in BR "Linked Project" should display the number of linked project.

Rest looks OK!

**Comment by elicoidal** [comment]
@victormartinelicocorp

File: business_requirement_deliverable_project/models/business.py [hunk 1]
```diff
@@ -43,6 +44,22 @@ def generate_projects_wizard(self):
         compute='_compute_planned_hour'
     )
 
+    @api.multi
+    def open_linked_projects(self):
```
Comment (martintorresvm) [inline business_requirement_deliverable_project/models/business.py]: @darshan-serpent This returns type action so the name need to be according:
s/open_linked_projects
r/action_open_linked_projects

File: business_requirement_deliverable_project/models/project.py [hunk 2]
```diff
@@ -11,6 +11,39 @@ class Project(models.Model):
...
+    )
+
+    business_requirement_deliverable_id = fields.Many2one(
+        comodel_name='business.requirement.deliverable',
+        string='Business Requirement Deliverable',
+        help='Link the Project and the business requirement deliverable',
+    )
+
+    @api.multi
+    def open_linked_br_dl(self):
```
Comment (martintorresvm) [inline business_requirement_deliverable_project/models/project.py]: @darshan-serpent This returns type action so the name need to be according:
s/open_linked_br_dl
r/action_open_linked_br_dl

**Review by martintorresvm — CHANGES_REQUESTED** [review]
Others LGTM

**Comment by YogeshMahera-SerpentCS** [comment]
@victormartinelicocorp could you please review.

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS Can you recheck Travis?

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal Travis Green

**Comment by elicoidal** [comment]
@darshan-serpent I still see many issues with smart buttons in Project and the more I think about the more I think this is anyway a mistake create a smart button for this kind of field: it should be a simple m2o field right (for one given project, there could only be one BR, right?) ? If so I would put it below customer in the project form
cc @victormartinelicocorp @sudhir-serpentcs

**Comment by elicoidal** [comment]
@sudhir-serpentcs any feedback on this?

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal Could you please review

**Review by elicoidal — APPROVED** [review]
@YogeshMahera-SerpentCS LGTM!

**Comment by elicoidal** [comment]
@victormartinelicocorp @dreispt @pedrobaeza Thanks for your review

File: business_requirement_deliverable_project/models/business.py [hunk 3]
```diff
@@ -42,6 +43,36 @@ def generate_projects_wizard(self):
...
     )
+    linked_project_count = fields.Integer(
+        compute='_compute_linked_project_count',
+        string="Number of Business Requirements"
+    )
+
+    @api.depends('linked_project', 'deliverable_lines')
+    def _compute_linked_project_count(self):
+        for rec in self:
+            domain = ['|',
```
Comment (martintorresvm) [inline business_requirement_deliverable_project/models/business.py]: @darshan-serpent @YogeshMahera-SerpentCS Need to understand the exactly purpose of this domain?
This two fields:
`business_requirement_id`
`business_requirement_deliverable_id`

As parameters domains means should be fields on the `business.requirement` model, right?

**Review by martintorresvm — APPROVED** [review]
thanks @darshan-serpent @sudhir-serpentcs

**Comment by martintorresvm** [comment]
@elicoidal @seb-elico As you know we could see just the code that changes when it's created a PR. 

@YogeshMahera-SerpentCS Please it's very important to be sure and understand the scope. So if the topic is `Modify projects and BR id relationship` how comes that still exists `linked_project` fields on BR and BRD?

Furthermore on wizard for generate projects still exist check and assignations that depends on `linked_project`. 

Others smalls details, that could be quick correct.

Now we have and issue #241 and now I have to review the whole code and check whether is new or old modification.

**Pull request merged**